### PR TITLE
Removing MarkDuplicatesGATK

### LINKED ISSUE #4896: Retire the MarkDuplicatesGATK tool (for now)
@jamesemery has found that he's been able to make `MarkDuplicatesSpark` call directly into Picard code for much core functionality. As a result, `MarkDuplicatesGATK` has gotten out-of-date relative to both the Spark version and the Picard version.

I think that we should remove `MarkDuplicatesGATK` for now, as it's likely just confusing users. Longer term, we should still discuss migrating development of the Picard `MarkDuplicates` tool to GATK in order to avoid constant divergence between that version and the Spark version, but if we end up doing this we'd likely want to re-port from scratch anyway rather than use `MarkDuplicatesGATK` as a starting point.

### PR DESCRIPTION
**Pull request by jamesemery**
In light of the newly updated MarkDuplicatesSpark which is finally reaching the point where we trust it, we have evaluated that the work of maintaining both this tool and MarkDuplicatesSpark was too great considering how little code they could directly reuse relative to eachother. 

Resolves #4896 #3705

### CODE REVIEW

**Comment by jamesemery** [comment]
I did not remove EstimateLibraryComplexityGATK, It would be easy to do as part of this pr and would mean most of the markDuplicatesPackage can be removed entirely.

**Comment by lbergelson** [comment]
I'd kill EstimateLibraryComplexity while you're at it.

File: src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/MarkDuplicatesSparkIntegrationTest.java [hunk 1]
```diff
@@ -68,278 +48,39 @@ public void testMDOrder(final File input, final File expectedOutput) throws Exce
...
-                {"RUNID:7:1203:2886:37756", "RUNID:7:1203:2886:36756"},
-                {"RUNID:7:1203:2886:38756", "RUNID:7:1203:2886:36756"},
-
-                //Added a test for tiebreaking accounting for the short casting done in picard
-                {"HK3T5CCXX160204:3:1112:11586:37067", "HK3T5CCXX160204:3:1112:11586:32144"}
-        };
-    }
-
-    @Test(dataProvider = "readNameData")
-    public void testOpticalDuplicateClusterSamePositionNoOpticalDuplicates(final String readName1, final String readName2) {
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/MarkDuplicatesSparkIntegrationTest.java]: Why move all these tests up to AbstractMarkDuplicatesCommandLineTest?  It doesn't seem like things like `testHashCollisionHandling and `testMarkDuplicatesSparkIntegrationTestLocal` belong there.

**Comment by jamesemery** [comment]
@lbergelson I agree with you about moving tests up, though I would make the point that there were many tests that got pushed down in the first place because they would involve fixing bugs in MarkDuplicatesGATK that were already fixed in MarkDuplicatesSpark, I will audit the ones I did and didn't push up so i'm more confident there is a reason to have tests pushed down or not.

**Comment by jamesemery** [comment]
@lbergelson back to you

**Pull request merged**